File: 1_100/21.js
/**
 * Definition for singly-linked list.*/
  function ListNode(val, next) {
      this.val = (val===undefined ? 0 : val)
      this.next = (next===undefined ? null : next)
  }

/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
 var mergeTwoLists = function(list1, list2) {
     let returnList = new ListNode();
     let returnHead = returnList;

    while(list1 !== null || list2 !== null) {
         if(list1 === null) {
             returnList.next = new ListNode(list2.val);
             returnList = returnList.next;
             list2 = list2.next;
         } else if(list2 === null) {
             returnList.next = new ListNode(list1.val)
             returnList = returnList.next;
             list1 = list1.next;
         } else if(list1.val < list2.val) {
             returnList.next = new ListNode(list1.val);
             returnList = returnList.next;
             list1 = list1.next;
         } else if(list2.val < list1.val) {
             returnList.next = new ListNode(list2.val);
             returnList = returnList.next;
             list2 = list2.next;
         } else if(list1.val === list2.val) {
             returnList.next = new ListNode(list1.val);
             returnList = returnList.next;
             list1 = list1.next;
             returnList.next = new ListNode(list2.val);
             returnList = returnList.next;
             list2 = list2.next;
         }
     }

     return returnHead.next;
    
};

let list1 = new ListNode();
let list2 = new ListNode(0,null);

console.log(mergeTwoLists(list1, list2));

File: 1_100/21.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
 public class Solution
    {
        public ListNode MergeTwoLists(ListNode list1, ListNode list2)
        {       
            ListNode dummy = new ListNode(0);
            ListNode prev = dummy;


            while(list1 != null && list2 != null)
            {
                if(list1.val <= list2.val)
                {
                    prev.next = list1;
                    list1 = list1.next;                    
                } else
                {
                    prev.next = list2;
                    list2 = list2.next;                    
                }

                prev = prev.next;
            }

            prev.next = list1 == null ? list2 : list1;

            return dummy.next;

        }
    }